default/scripting/buildings/colonies/SP_BANFORO.focs.txt

```diff
@@ -45,18 +45,21 @@ BuildingType
         Population high = 0
         Not Planet environment = Uninhabitable species = "SP_BANFORO"
         Not Contains Building name = "BLD_COL_BANFORO"
-        ResourceSupplyConnected empire = Source.Owner condition = And [
-            Planet
-            OwnedBy empire = Source.Owner
-            Or [
-                And [
-                    Species name = "SP_BANFORO"
-                    Population low = [[MIN_RECOLONIZING_SIZE]]
-                    Happiness low = 5
-                ]
-                And [
-                    HasSpecial name = "EXTINCT_BANFORO_SPECIAL"
-                    Contains Building name = "BLD_XENORESURRECTION_LAB"
+        Or [
+            OwnerHasTech name = "TECH_COL_BANFORO"
+            ResourceSupplyConnected empire = Source.Owner condition = And [
+                Planet
+                OwnedBy empire = Source.Owner
+                Or [
+                    And [
+                        Species name = "SP_BANFORO"
+                        Population low = [[MIN_RECOLONIZING_SIZE]]
+                        Happiness low = 5
+                    ]
+                    And [
+                        HasSpecial name = "EXTINCT_BANFORO_SPECIAL"
+                        Contains Building name = "BLD_XENORESURRECTION_LAB"
+                    ]
```

Comment:
TY, I will update and test Exobots accordingly.

For extinct species I envisioned that building a lab on ruins grants the knowledge to reconstruct the species without any other resource requirement from the ruins.  In that case, I probably should have removed the `And` block here, resulting in:

```
Or [
    OwnerHasTech name = "TECH_COL_BANFORO"
    ResourceSupplyConnected empire = Source.Owner condition = And [
        Planet
        OwnedBy empire = Source.Owner
        Species name = "SP_BANFORO"
        Population low = [[MIN_RECOLONIZING_SIZE]]
        Happiness low = 5
    ]
]
```

Regardless of which method is desired, I should probably add some info to pedia articles reflecting the special requirements compared to a normal species.